### PR TITLE
update targeting for annex-b String.prototype methods based on MDN browser support tables

### PR DESCRIPTION
**Pull request by JakeChampion**
They all have the nearly same support which is why I have done them together

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/strike#Browser_compatibility

### CODE REVIEW

File: polyfills/String/prototype/strike/config.toml [hunk 1]
```diff
@@ -5,22 +5,21 @@ dependencies = [
...
-# opera = "*"
-op_mini = "*"
-# op_mob = "*"
-# safari = "*"
-firefox_mob = "< 4"
-# samsung_mob = "*"
+android = "<1"
+bb = "<1"
+chrome = "<1"
+edge = "<1"
```
Comment (notlee) [inline polyfills/String/prototype/strike/config.toml]: 12?

File: polyfills/String/prototype/anchor/config.toml [hunk 2]
```diff
@@ -5,22 +5,20 @@ dependencies = [
...
-op_mini = "*"
-# op_mob = "*"
-# safari = "*"
-firefox_mob = "< 4"
-# samsung_mob = "*"
-
+android = "<1"
+bb = "<1"
+chrome = "<1"
+edge = "<1"
```
Comment (notlee) [inline polyfills/String/prototype/anchor/config.toml]: 12? ... according to mdn... but I feel like these features would have been supported for forever

Comment (JakeChampion) [reply polyfills/String/prototype/anchor/config.toml]: 12 is the first version of edge but i'll update anyway

Comment (notlee) [reply polyfills/String/prototype/anchor/config.toml]: I did not know this

**Pull request merged**